include/srf/segment/builder.hpp

```diff
@@ -73,31 +74,49 @@ class Builder final
     template <typename SourceTypeT, typename CreateFnT>
     auto make_source(std::string name, CreateFnT&& create_fn)
     {
-        return make_object(std::move(name),
-                           std::make_unique<node::RxSource<SourceTypeT>>(
-                               rxcpp::observable<>::create<SourceTypeT>(std::forward<CreateFnT>(create_fn))));
+        auto uptr = std::make_unique<node::RxSource<SourceTypeT>>(
+            rxcpp::observable<>::create<SourceTypeT>(std::forward<CreateFnT>(create_fn)));
+
+        auto trace_stats = benchmarking::TraceStatistics::get_or_create(name);
+        uptr->source_add_watcher(trace_stats);
+
+        return make_object(std::move(name), std::move(uptr));
```

Comment:
I'm not sure I agree. Its not clear that all ObjectT's are or should be 'Nodes', or are some form of Source/Sink.

In `make_object`, unless we just try random casting, we have no idea what ObjectT is to even test if it has a source/sink. Adding the watchers to the node is really what we want; I think it would be better if this wasn't duplicated between c++/python paths, but I'm reasonably ok with where it is. 

So far as implementing it on the Watcher object, that seems awkward, because Watchers don't have any notion of a unique name.

Reply:
Applied the changes we discussed. All make_xxx elements on the builder are now just proxy calls to construct_object; construct_object automatically detects if the ObjectT its creating has the RxSource/Sink API calls to add watchers and if so it attaches them.